arch/arc/core/switch.S

```diff
@@ -113,7 +97,7 @@ SECTION_FUNC(TEXT, __swap)
 	/* sflag instruction is not supported in current ARC GNU */
 	.long 0x00ff302f
 #else
-	/* disable stack checking during swap */
+	/* disable stack checking during switch */
```

Comment:
Is it really necessary to disable stack checking here? If required, can it moved past the next push and save_callee_saved_regs?

Reply:
yes, it required here because in the coming 	_save_callee_saved_regs and _load_callee_saved_regs, sp will change. If not disabled here, stack checking exception will raise

Reply:
for the load I understand, but the save doesn't change the sp other than allocating space for saving the registers, does it?
The loading is less sensitive, but the save could write out of bounds.
Maybe the save/load macro's need some work to get the checking more tight. If so, a separate issue and PR for that would be better than solving it in this PR.

Reply:
@ruuddw , you are right,  I already made a fix. So if no other issues, i think it can be merged